### PR TITLE
prov/gni: Allow ctx enable before bind of corresponding rx/tx ctx.

### PR DESCRIPTION
**Pull request by chuckfossen**
Fixes ofi-cray/libfabric-cray#1052

Signed-off-by: Chuck Fossen <chuckf@cray.com>

@sungeunchoi @hppritcha

### CODE REVIEW

File: prov/gni/test/sep.c [hunk 1]
```diff
@@ -222,6 +222,11 @@ void sep_setup(void)
 			ret = fi_enable(tx_ep[i][j]);
 			cr_assert(!ret, "fi_enable");
 
+			/* test if bind fails */
+			ret = fi_ep_bind(tx_ep[i][j], &tx_cq[i][j]->fid,
+					 FI_TRANSMIT);
+			cr_assert(ret, "fi_ep_bind should fail");
```
Comment (sungeunchoi) [inline prov/gni/test/sep.c]: Same comment I gave to Howard in his PR.  This should be a separate test, not in every test.

Comment (chuckfossen) [reply prov/gni/test/sep.c]: ok

Comment (chuckfossen) [reply prov/gni/test/sep.c]: Is it really worth adding another setup routine to check this?

Comment (sungeunchoi) [reply prov/gni/test/sep.c]: No.  Just add another test that uses the same set up and tries to bind a new tx or rx cq to the existing, already enabled eps.

**Comment by chuckfossen** [comment]
I updated the critereon test to test the fail paths conditionally.

File: prov/gni/test/sep.c (hunk 1 above)
Comment (chuckfossen) [reply prov/gni/test/sep.c]: Oh, yes that makes more sense.

**Comment by sungeunchoi** [comment]
👍

**Pull request merged**